### PR TITLE
Luxury shuttle improvements

### PR DESCRIPTION
**Pull request by cacogen**
## About The Pull Request

Competing pull request with #48382

- Adds ticket booth between coach and first class for hijackers
- Widens coach by one tile and adds chairs
- Adds shuttle brig, free to enter, still costs to get into first class
- Makes it so boarding cost is mentioned when luxury shuttle purchase is announced
- Makes it so shuttle names are announced when they arrive as another warning you need to pay for this one
- Adds interior windows to coach, with shutters operated from first class
- Moves bathroom for shuttle brig
- Changes ticket forcewalls to scanner gates, function the same but buzz and turn red when someone can't pass and turn blue when they do
- Bunch of other aesthetic changes seen below

Old:

New:

deleted an airlock to Steerage in screenshot above to show scanner gate used as new forcefield

There are other minor changes:

~~- Fixes some sink and coastline corner sprites so they rotate correctly with shuttle -- will cause some sprite directions to be wrong elsewhere~~
- Adds indestructible diamond walls
- Fixes runtime where paying with cash and needing change didn't dispense any

## Why It's Good For The Game

Addresses most of the issues with the shuttle raised in #48382

## Changelog
:cl: cacogen
tweak: Luxury shuttle is a little more sophisticated
/:cl:

### CODE REVIEW

**Comment by TheVekter** [comment]
MapDiffBot is still fucked, post a screenshot.

**Comment by Bawhoppen** [comment]
In my opinion, this would be acceptable if you make steerage 2 rows bigger (no shuttle should be insanely punishing), and make it look nicer/more cohesive in some way. Most importantly, I believe the the ticket booths to need to be changed to something more obvious and reasonable than magical circles (we have metal detectors right? maybe slightly tweak one of those sprites)

**Comment by Ziiro** [comment]
Idea: the barrier between first class and steerage is some serious high grade plastitanium glass or something. To laugh at the poors.

Although that would allow lasers. Hmm.

**Comment by cacogen** [comment]
@Bawhoppen

>make steerage 2 rows bigger (no shuttle should be insanely punishing)

done, also added cramped chairs

>make the whole thing look nicer/more cohesive in some way

i'm gonna look at palette sites to try to get some colours complimenting the gold/diamond. other than that, can't really do much for you. i've never been good at aesthetics

>I believe the the ticket booths to need to be changed to something more obvious and reasonable than magical circles

i've wanted to make them the metal detector type things we have for awhile that i think you're also referring to (i.e. laser field that turns red and buzzes if someone passes who's a lizard, hungry, diseased, etc) , so i'll look at that

**Comment by Bawhoppen** [comment]
@cacogen okay cool, that's personally good in my opinion

**Comment by Iatots** [comment]
The poor section now is just as capacious as your typical shuttle now.

**Comment by CAPTTLasky** [comment]
That poor section needs a downgrade but otherwise I like this.

**Comment by Ziiro** [comment]
No actual floors, and maybe just give them rolling office chairs

**Comment by HomocidlePancakes** [comment]
I like the low class segment. Looks like your average budget airline.

**Comment by Iatots** [comment]
I think there are benches in the code. That would be better, along with a smaller compartment.

**Comment by carlarctg** [comment]
poor section is too nice

add shutters with centcommlike windows to the walls surrounding it, and give the first class a button to control them. laugh at the peasants

**Comment by CRITAWAKETS** [comment]
the poor do not deserve diamond walls, give em sandstone

**Comment by cacogen** [comment]
@HomocidlePancakes
i liked it too, for that same reason

@carlarctg 
it's uglier now without floor tiles, and i added tlal's suggestion of shutters using the centcom type windows you mentioned that are actually transparent indestructible undeconstructible walls with window sprites

@Iatots 
benches don't fit with a shuttle, nor do any of the other shitty looking seats, steerage is now smaller as i agreed there were too many chairs

@CRITAWAKETS 
i thought about hiding the walls using tinted directional windows, but it looked ugly. i'm not going to give them their own cheaper walls because i'm not double layering walls which means they'll have to share a wall at some position anyway and the colours won't work

@Ziiro 
i gave them plating because people were/are complaining it looked "too nice", though i preferred plastic myself. office chairs don't make sense on a shuttle and are ugly in the wrong sense. if they all slammed into each other when it took off i'd reconsider

@CAPTTLasky
it is downgraded. it now looks like shit, though more ugly mapping than shitty shuttle in my opinion

**Comment by Ziiro** [comment]
I'm just nitpicking and i think everything you've done here is actually great by the way.

If the office chairs slammed into each other on take off that would be funny as shit but I remember the series of PRs exploring why that can't be so I understand

**Comment by Bawhoppen** [comment]
I think most people are being influenced by a status quo bias with just really how small steerage actually is. I think an actually reasonable one would encompass the entire back 1/3rd of the shuttle (rightfully near the engines, of course). The two different cabins should really mainly differ in quality though and feeling cramped, rather than having a size which is actively punishing. Of course, the price to enter the upper class could also be increased as well, naturally. I'll try to make another alternative version proposal.

**Comment by Iatots** [comment]
The original poor section being an afterthought was pottery.

Also there is nothing wrong with shuttles being punishing, the problem is people using them too often, which can be fixed by adjusting price and requirements.

And as for the status quo bias argument: With how salaries are currently implemented, there never was an easier time getting on the shuttle's good  section; The only ones that need to cope are assistant mains, and even then those with 2 braincells can just tide 500 bucks off any doctor's ID since they are not PIN protected.

**Comment by Bawhoppen** [comment]
@Iatots I am going to have to fundamentally disagree with you on that *emergency* shuttles should ever be seriously punishing. In my opinion, it's neither good flavor (which I think is very important to the game experience), nor enjoyable gameplay. 

Also, the fact that it's easy to get into the first class section is kinda backwards logic in my view. Shouldn't that be where only the minority of rich people deserve to get into, with most people having to go economy-class? Seems like that would be the fairest and most fun way to do it the whole gimmick, I think.

**Review by kriskog — CHANGES_REQUESTED** [review]
> Fixes some sink and coastline corner sprites so they rotate correctly with shuttle -- will cause some sprite directions to be wrong elsewhere

How widespread is "wrong elsewhere"? Also, sink sprites at least are still weird after shuttle rotation. You may need  to think up something else here.

Other than that, changes look good and this is preferable to removal.

**Comment by Bawhoppen** [comment]
@cacogen In the process of making another modified version, I've discovered you seemed to have accidentally used falsewalls instead of actual walls for the entire shuttle. 
Also, while I'm mentioning things, the ticket gates should really be on a different tile than the airlocks, rather than the same one (like the next tile inside of the shuttle)

**Comment by cacogen** [comment]
If there are false walls instead of the new indestructible diamond ones, you have an old version

File: code/game/machinery/computer/communications.dm [hunk 1]
```diff
@@ -169,7 +169,7 @@
 							SSshuttle.existing_shuttle = SSshuttle.emergency
 							SSshuttle.action_load(S)
 							D.adjust_money(-S.credit_cost)
-							minor_announce("[usr.real_name] has purchased [S.name] for [S.credit_cost] credits." , "Shuttle Purchase")
+							minor_announce("[usr.real_name] has purchased [S.name] for [S.credit_cost] credits.[S.name == "Luxury Shuttle" ? " This shuttle costs 500 credits to board." : ""]" , "Shuttle Purchase")
```
[the remark this answers is not in the document]
Comment (cacogen) [reply code/game/machinery/computer/communications.dm]: done

File: code/modules/shuttle/special.dm [hunk 2]
```diff
@@ -335,10 +346,12 @@
 		for(var/obj/I in counted_money)
 			qdel(I)
 		if(!check_times[AM] || check_times[AM] < world.time) //Let's not spam the message
-			to_chat(AM, "<span class='notice'>$[payees[AM]] received. You need $[threshold-payees[AM]] more.</span>")
+			to_chat(AM, "<span class='notice'>[payees[AM]]N received. You need [threshold-payees[AM]]N more.</span>")
```
[the remark this answers is not in the document]
Comment (cacogen) [reply code/modules/shuttle/special.dm]: fixed

**Comment by cacogen** [comment]
>How widespread is "wrong elsewhere"? Also, sink sprites at least are still weird after shuttle rotation. You may need to think up something else here.

The directions on those sprites are wrong in two places. They need to be swapped. In fixing them though, anywhere that type of sink or that coastline corner at one of those directions will change to face the wrong way. I've removed the changed from this PR to atomise it.

**Review by kriskog — APPROVED** [review]

**Pull request merged**